### PR TITLE
Updated service pool to only add new node as usable if the node passes an initial healthcheck first.

### PR DESCRIPTION
**Pull request by bivasdas**

### CODE REVIEW

File: core/src/main/java/com/bazaarvoice/ostrich/pool/ServicePool.java [hunk 1]
```diff
@@ -429,10 +429,45 @@ void forceHealthChecks() {
     }
 
     private synchronized void addEndPoint(ServiceEndPoint endPoint) {
-        _serviceCache.register(endPoint);
-        _recentlyRemovedEndPoints.remove(endPoint);
-        _badEndPoints.remove(endPoint);
-        LOG.debug("End point added to service pool. End point ID: {}", endPoint.getId());
+        // submit calls schedule with 0 delay
+        _healthCheckExecutor.submit( new EndPointAdditionProcessor( endPoint ) );
```
Comment (ohhatiya) [inline core/src/main/java/com/bazaarvoice/ostrich/pool/ServicePool.java]: I don't understand why this is necessary. Can't it just remove the end point from recentlyRemovedEndPoints and then call markEndPointAsBad?

**Comment by ohhatiya** [comment]
Can this be closed in favor of #181?

File: core/src/main/java/com/bazaarvoice/ostrich/pool/ServicePool.java (hunk 1 above)
Comment (bivasdas) [inline core/src/main/java/com/bazaarvoice/ostrich/pool/ServicePool.java]: yes. Looks like I missed that clue.

**Comment by bivasdas** [comment]
Yes.

**Pull request closed without merging**